octavia_f5/controller/worker/tasks/f5_tasks.py

```
@@ -158,29 +169,48 @@ class EnsureRouteDomain(task.Task):
-            path = f"/mgmt/tm/net/route-domain/net-{network.id}"
-            device_response = bigip.get(path=path)
-
         # Create route_domain if not existing
-        if device_response.status_code == 404:
+        if existing_route_domain is None:
             res = bigip.post(path='/mgmt/tm/net/route-domain', json=rd)
             res.raise_for_status()
             return res.json()
 
-        device_rd = device_response.json()
-        if device_rd.get('vlans', []) != vlans:
-            res = bigip.patch(path=f"/mgmt/tm/net/route-domain/{device_rd['fullPath']}",
+        if existing_route_domain.get('vlans', []) != vlans:
+            res = bigip.patch(path=f"/mgmt/tm/net/route-domain/{existing_route_domain['fullPath']}",
                               json={'vlans': vlans})
             res.raise_for_status()
             return res.json()
-        return device_rd
+
+        return existing_route_domain
+
+    @decorators.RaisesIControlRestError()
+    def revert(self, network: f5_network_models.Network,
+               bigip: bigip_restclient.BigIPRestClient,
+               existing_route_domain, *args, **kwargs):
+        paths = [
+            f"/mgmt/tm/net/route-domain/vlan-{network.vlan_id}",
+            f"/mgmt/tm/net/route-domain/net-{network.id}"
+        ]
+        if existing_route_domain is not None:
+            LOG.warning(f"Reverting EnsureRouteDomain: Not deleting RouteDomain, since it existed before "
+                        f"the task was run: {existing_route_domain}")
+            return
+
+        """ Task to delete Route Domain """
+        res = None
+        for path in paths:
+            if bigip.get(path=path).ok:
+                res = bigip.delete(path=path)
+                break
+
+        if res and not res.ok:
+            LOG.warning("%s: Failed removing route domain for network_id=%s vlan_id=%s: %s",
+                        bigip.hostname, network.id, network.vlan_id, res.content)
```

Please reraise the error, like this: `res.raise_for_status()`.